src/python/WMCore/MicroService/MSPileup/MSPileupMonitoring.py

```diff
@@ -0,0 +1,185 @@
+"""
+File       : MSPileupMonitoring.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupMonitoring provides bridge between MSPileup
+             service and CMS Monitoring infrastructure
+
+Usage:
+    # initialize monitoring manager
+    monMgr = MSPileupMonitoring()
+    # start monitoring trhead
+    self.monMgr.startMonThread()
+    # check monitoring thread status
+    status = self.monMgr.isAlive()
+    # restart monitoring thread
+    self.monMgr.restart()
+"""
+
+# system modules
+import time
+import datetime
+
+# WMCore modules
+from WMCore.MicroService.MSCore.TaskManager import start_new_thread
+from WMCore.MicroService.MSCore.MSManager import daemonOpt
+from WMCore.MicroService.MSPileup.MSPileupData import MSPileupData
+
+# CMSMonitoring modules
+from CMSMonitoring.StompAMQ7 import StompAMQ7 as StompAMQ
+
+
+# THR_FLAT controls kill process of daemon
+THR_FLAG = False
+
+
+def flatDocuments(doc):
+    "Helper functino to flat out MSPileup document"
+    docs = flatKey(doc, 'ruleIds')
+    docs = (f for d in docs for f in flatKey(d, 'currentRSEs'))
+    docs = (f for d in docs for f in flatKey(d, 'expectedRSEs'))
+    docs = (f for d in docs for f in flatKey(d, 'campaigns'))
+    for doc in docs:
+        yield doc
+
+
+def flatKey(doc, key):
+    "Helper function to flat out values of given key in a document"
+    for item in doc[key]:
+        ndoc = dict(doc)
+        nkey = key[:-1]  # ruleIds -> ruleId
```

Comment:
Alan, apart from doc string typo, I do not understand what you mean "wrong". I decided to flatten the document from and replace keys with plural name to single, e.g. "ruleIds" which is a list will become "ruleId" which will have single value. I tested this functionality in provided unit test. This will correctly reflect the data-type, since "rulesIds" is a list data type, while when we will flatten the dict we will send only one element of such list, i.e. in this case a string. I just need to understand what you meant by "wrong" and I can adjust accordingly.

Please note, to simplify queries in ES and grafana dashboards, it is desired to flatten the document. Since we have few lists, I thought it will be natural to use plural to single conversion, e.g. "ruleIds" will become "ruleId, "currentRSEs" will become "currentRSE", etc.

Reply:
Ahh, now I understand what you have done in the code. Yes, you are right.
It might be better to add a comment in the line above though saying something like:
```
# convert plural to singular key name
```

About the docstring, please read: https://github.com/dmwm/WMCore/blob/master/CONTRIBUTING.rst#project-docstrings-best-practices and adjust the code accordingly.

Reply:
ok